src/components/Toggle/Toggle.js

```diff
@@ -60,30 +60,36 @@ const ToggleWrapper = styled('div')`
 /**
  * A toggle component with support for labels and additional explanations.
  */
-const Toggle = ({ label, explanation, noMargin, ...props }) => (
-  <ToggleWrapper {...{ noMargin }}>
-    <Switch {...props} />
-    {label &&
-      explanation && (
-        <ToggleTextWrapper>
+const Toggle = ({ label, explanation, noMargin, ...props }) => {
+  const switchId = uniqueId('toggle-switch_');
+  const labelId = uniqueId('toggle-label_');
+  return (
+    <ToggleWrapper {...{ noMargin }}>
+      <Switch {...props} aria-labelledby={labelId} id={switchId} />
+      {(label || explanation) && (
+        <ToggleTextWrapper id={labelId} htmlFor={switchId}>
           {label && (
-            <ToggleLabel element="label" size={Text.KILO}>
+            <ToggleLabel size={Text.KILO} noMargin>
               {label}
             </ToggleLabel>
           )}
           {explanation && (
-            <ToggleExplanation size={Text.KILO}>
+            <ToggleExplanation size={Text.KILO} noMargin>
               {explanation}
             </ToggleExplanation>
           )}
         </ToggleTextWrapper>
       )}
-  </ToggleWrapper>
-);
+    </ToggleWrapper>
+  );
+};
 
 Toggle.propTypes = {
   label: PropTypes.string,
```

Comment:
Oupsies. Did I never add these? 😞 Would you mind adding them?

Reply:
Sure thing!